lib/nrf_modem_lib/nrf91_sockets.c

```diff
@@ -100,9 +100,11 @@ static void nrf_to_z_ipv4(struct sockaddr *z_out,
 {
 	struct sockaddr_in *ptr = (struct sockaddr_in *)z_out;
 
-	ptr->sin_port = nrf_in->sin_port;
-	ptr->sin_family = AF_INET;
-	ptr->sin_addr.s_addr = nrf_in->sin_addr.s_addr;
+	if (ptr) {
```

Comment:
Curious, `nrf_to_z_ipv4()` is a file scoped function, and it's being called here:
https://github.com/nrfconnect/sdk-nrf/blob/dd30adb7cabed392ccbb73a49d91305f4d06dbad/lib/nrf_modem_lib/nrf91_sockets.c#L273
and input is tested for `NULL` here:
https://github.com/nrfconnect/sdk-nrf/blob/dd30adb7cabed392ccbb73a49d91305f4d06dbad/lib/nrf_modem_lib/nrf91_sockets.c#L268-L269

and here:
https://github.com/nrfconnect/sdk-nrf/blob/dd30adb7cabed392ccbb73a49d91305f4d06dbad/lib/nrf_modem_lib/nrf91_sockets.c#L343-L344
tested for `NULL` right above:
https://github.com/nrfconnect/sdk-nrf/blob/dd30adb7cabed392ccbb73a49d91305f4d06dbad/lib/nrf_modem_lib/nrf91_sockets.c#L340

and here:
https://github.com/nrfconnect/sdk-nrf/blob/dd30adb7cabed392ccbb73a49d91305f4d06dbad/lib/nrf_modem_lib/nrf91_sockets.c#L570

tested for `NULL`:
https://github.com/nrfconnect/sdk-nrf/blob/dd30adb7cabed392ccbb73a49d91305f4d06dbad/lib/nrf_modem_lib/nrf91_sockets.c#L545-L548

so i'm surprised (and concerned) if you have experienced the input field being `NULL`, as that could indicate another problem with the code.

Reply:
@tejlmand Good point, reverting `nrf_to_z_ipv4` to original implementation.

Reply:
> @tejlmand Good point, reverting `nrf_to_z_ipv4` to original implementation.

@MirkoCovizzi  When changing back to original implementation, could you please check if the commit message is still up-to-date ?

Reply:
@tejlmand commit message is not up to date, I need to update it because a lot changed since then.